crates/locutus-node/src/operations/subscribe.rs

```diff
@@ -54,9 +55,27 @@ impl StateMachineImpl for SubscribeOpSM {
                     retries: 0,
                 })
             }
-            (SubscribeState::ReceivedRequest, SubscribeMsg::SeekNode { .. }) => {
+            (SubscribeState::ReceivedRequest, SubscribeMsg::SeekNode { found: true, .. }) => {
                 Some(SubscribeState::Completed)
             }
+            (
+                SubscribeState::ReceivedRequest,
+                SubscribeMsg::SeekNode {
+                    target,
+                    retries,
+                    found: false,
+                    ..
+                },
+            ) => {
+                if retries < &SubscribeOp::MAX_RETRIES {
+                    Some(SubscribeState::AwaitingResponse {
+                        skip_list: vec![],
+                        retries: 0,
```

Comment:
is this correct? shouldn't it `retries: retries - 1`

Reply:
right, it should be retires + 1

Reply:
I am not sure if the number of retries needs to be updated in this case.

Reply:
sorry, meant `retries + 1` but yes I don't think has to be updated here. but then should be equal to the last  `retries`not 0, and it will be updated inside the machine.